phase1-coordinator/src/coordinator_state.rs

```diff
@@ -2517,6 +2527,50 @@ impl CoordinatorState {
             .collect()
     }
 
+    /// This will drop a participant if it hasn't accepted a task for more than
+    /// [crate::environment::Environment]'s `participant_acceptance_timeout`.
+    fn update_participant_accepted_drops(
+        &mut self,
+        time: &dyn TimeSource,
+    ) -> Result<Vec<DropParticipant>, CoordinatorError> {
+        let participant_acceptance_timeout = self.environment.participant_acceptance_timeout();
+        let now = time.utc_now();
+
+        self.current_contributors
+            .clone()
+            .iter()
+            .chain(self.current_verifiers.clone().iter())
+            .filter_map(|(participant, participant_info)| {
+                // Ensure we get the proper elapsed time since the last task was assigned.
+                // If no tasks have been assigned yet, we measure from the starting time.
+                let last_seen = if participant_info.last_released.is_none() {
+                    participant_info.first_seen
+                } else {
+                    participant_info.last_released.unwrap()
```

Comment:
if there's a way to avoid panic here, it would be better in terms of security

Reply:
It should already be guaranteed because we check `is_none`, but I guess an `unwrap_or_default` can't hurt!

Reply:
I though these were different variables, my bad, was confused by the mix of last_released and first_seen. Yes, no harm in unwrap after is_none check.